woocommerce/woocommerce-android
# Prevent scan on configurable products

Closes: #9541

### Why
Configurable products require a configuration before adding the item to the order. Products scanned are directly added to the item list, so we are restricting those products from being scanned to prevent creating a configurable product without a configuration.

### Description
This PR adds configurable products to the barcode-restricted products. When we try to scan a configurable product, the app will show a message indicating that these products are restricted.

### Testing instructions

#### Pre-requisites
1. Install the product bundles extensions
2. Create two bundle products, one that needs a configuration and other that doesn't
3. Assign a numeric SKU for those two products and create a barcode (you can use [this site](https://barcode.tec-it.com/en/UPCA?data=22345678999) for creating the barcode)

#### Testing
TC1
1. Open the orders tab
2. Tap on create new order (+)
3. Tap on the scan icon to the left of the add products button
4. Scan the barcode generated for the product that needs configuration
5. Check that the shows an error message

TC2
1. Open the orders tab
2. Tap on create new order (+)
3. Tap on the scan icon to the left of the add products button
4. Scan the barcode generated for the product that doesn't need configuration
5. Check that the bundle product is added to the list of items

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/18119390/7ed2bd46-a14b-444c-bff9-ae9e7fa6110b
